FinSearchUnified/Tests/Bundle/SearchBundleFindologic/QueryBuilderFactoryTest.php

```diff
@@ -0,0 +1,180 @@
+        $criteria->addCondition(new IsAvailableCondition());
+
+        $query = $this->factory->createQuery($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertArrayHasKey('attrib', $params, 'Attributes were expected to be present in the parameters');
+        $attrib = $params['attrib'];
+
+        // CategoryCondition
+        $this->assertArrayHasKey('cat', $attrib, 'Category was expected to be present in the attribute parameters');
+        $this->assertEquals(
+            ['Genusswelten', 'Genusswelten_Tees und Zubehör_Tees'],
+            $attrib['cat'],
+            'Expected categories to contain the name of the provided category IDs'
+        );
+
+        // PriceCondition
+        $this->assertArrayHasKey('price', $attrib, 'Prices were expected to be present in the attribute parameters');
+        $this->assertEquals(1, $attrib['price']['min'], 'Expected minimum price to be 1');
+        $this->assertEquals(20, $attrib['price']['max'], 'Expected maximum price to be 20');
+
+        // ProductAttributeCondition
+        $this->assertArrayHasKey('vendor', $attrib, 'Expected "vendor" to be available in the attribute parameters');
+        $this->assertEquals(
+            ['Findologic Rockers'],
+            $attrib['vendor'],
+            'Expected vendor to be an array containing "Findologic Rockers"'
+        );
+
+        // SearchTermCondition
+        $this->assertArrayHasKey('query', $params, 'Expected search query to be in the parameters');
+        $this->assertSame('blubbergurke', $params['query'], 'Expected search query to be "blubbergurke"');
+
+        $this->assertCount(3, $attrib, 'Expected attributes to not contain any other parameters');
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function testCreateQueryWithoutSorting()
+    {
+        $criteria = new Criteria();
+
+        $query = $this->factory->createQueryWithSorting($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertArrayNotHasKey('order', $params, 'Sorting was not expected to be present in the parameters');
+        $this->assertArrayNotHasKey('attrib', $params, 'No attributes were expected to be present in the parameters');
+        $this->assertArrayNotHasKey('query', $params, 'No search query was expected to be present in the parameters ');
+    }
+
+    /**
+     * @dataProvider sortingProvider
+     *
+     * @param SortingInterface $sorting
+     * @param string $expected
+     *
+     * @throws Exception
+     */
+    public function testCreateQueryWithSingleSorting($sorting, $expected)
+    {
+        $criteria = new Criteria();
+        $criteria->addSorting($sorting);
+
+        $query = $this->factory->createQueryWithSorting($criteria, $this->context);
+        $params = $query->getParameters();
+
+        $this->assertSame($expected, $params['order'], sprintf('Expected sorting to be "%s"', $expected));
```

Comment:
This passes because of the way PHP works but doesn't accurately represent what we are trying to test. I would suggest that:
* If `$expected` is `null`:
  * Check that key `order` doesn't exist in `$params`.
* If `$expected` is not `null`:
  * Check for the value of `$params['order'] (already done).
  * Check that key `order` exists in `$params`.